### PR TITLE
Allow `EESSI-extend` as a loaded module when using EasyBuild

### PR DESCRIPTION
**Pull request by ocaisa**
Also edited the umask for various scenarios:
- for host wide installations, installations are world readable (umask 022)
- for project installations, installations are also group writable (umask 002)
- for user installations, installations are only accessible to the user (umask 077)

### CODE REVIEW

File: EESSI-extend-2023.06-easybuild.eb [hunk 1]
```diff
@@ -139,12 +139,17 @@ setenv ("EASYBUILD_SYSROOT", sysroot)
 setenv ("EASYBUILD_PREFIX", pathJoin(working_dir, "easybuild"))
 setenv ("EASYBUILD_INSTALLPATH", easybuild_installpath)
 setenv ("EASYBUILD_HOOKS", pathJoin(os.getenv("EESSI_PREFIX"), "init", "easybuild", "eb_hooks.py"))
-setenv ("EASYBUILD_UMASK", "002")
+
+-- Make sure to use the general umask that allows a global read
+setenv ("EASYBUILD_UMASK", "022")
+
+-- All this module to be loaded when running EasyBuild
```
Comment (bedroge) [inline EESSI-extend-2023.06-easybuild.eb:146]:
```suggestion
-- Allow this module to be loaded when running EasyBuild
```

File: easystacks/software.eessi.io/2023.06/rebuilds/20240506--eb-4.9.1-EESSI-extend-allow-loaded.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# 2024.05.15
+# The module is an EasyBuil created module and therefore needs to be an allowed
```
Comment (bedroge) [inline easystacks/software.eessi.io/2023.06/rebuilds/20240506--eb-4.9.1-EESSI-extend-allow-loaded.yml:2]:
```suggestion
# The module is an EasyBuild created module and therefore needs to be an allowed
```

**Comment by bedroge** [comment]
@ocaisa I guess this is ready to be (re)built?

**Comment by ocaisa** [comment]
@bedroge Yes, I think so. The controversial thing is whether these settings for the `umask` are reasonable? The user one is quite restrictive

**Comment by bedroge** [comment]
> @bedroge Yes, I think so. The controversial thing is whether these settings for the `umask` are reasonable? The user one is quite restrictive

@ocaisa Looks fine to me. We can explain in the docs what the defaults are and how you can override them manually?

**Review by bedroge — APPROVED** [review]

**Comment by bedroge** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512

**Comment by bedroge** [comment]
Hmm... I think we're hitting this issue again: https://github.com/EESSI/software-layer/issues/556. The directory seems empty and writable, but there's some stuff showing up if you check one of the old subdirectories:
```
bedroge@x86-64-amd-zen3-node2 ~ $ ls -la /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/EESSI-extend/2023.06-easybuild/    
total 0
drwxrwxr-x 2 bedroge bedroge 42 Jun  7 11:54 .
drwxr-xr-x 3 bedroge bedroge 31 Jun  7 11:54 ..
bedroge@x86-64-amd-zen3-node2 ~ $ ls -la /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/EESSI-extend/2023.06-easybuild/easybuild/
total 62
dr-xr-xr-x 3 bedroge bedroge  4096 May  7 13:41 .
drwxrwxr-x 2 bedroge bedroge    42 Jun  7 11:54 ..
-r--r--r-- 1 bedroge bedroge  2754 May  7 13:41 EESSI-extend-2023.06-easybuild-easybuild-devel
-r--r--r-- 1 bedroge bedroge  7976 May  7 13:28 EESSI-extend-2023.06-easybuild.eb
-r--r--r-- 1 bedroge bedroge 22875 May  7 13:41 easybuild-EESSI-extend-2023.06-20240507.134135.log.bz2
-rw-rw-r-- 1 bedroge bedroge 19051 May  7 13:41 easybuild-EESSI-extend-2023.06-20240507.134135_test_report.md
dr-xr-xr-x 4 bedroge bedroge  4096 May  7 13:41 reprod
```

**Comment by bedroge** [comment]
Added user write permissions to the EESSI-extend installation dirs, trying again...

**Comment by bedroge** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512

**Comment by bedroge** [comment]
Still some permission issues. Also recursively changed the ownership of all EESSI-extend directories now.

**Comment by bedroge** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3

**Comment by bedroge** [comment]
Still some permission issues :disappointed: 
```
easybuild.tools.build_log.EasyBuildError: "Failed to chmod/chown several paths: ['/cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/EESSI-extend/2023.06-easybuild/easybuild/easybuild-EESSI-extend-2023.06-20240507.134135.log.bz2', '/cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/EESSI-extend/2023.06-easybuild/easybuild/easybuild-EESSI-extend-2023.06-20240507.134135_test_report.md'] (last error: [Errno 13] Permission denied: '/cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/EESSI-extend/2023.06-easybuild/easybuild/easybuild-EESSI-extend-2023.06-20240507.134135_test_report.md')"
```

**Comment by bedroge** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512

**Comment by bedroge** [comment]
Deployed, and works as expected:
```
{EESSI 2023.06} $ ml EESSI-extend
-- Using /tmp/$USER as a temporary working directory for installations, you can
override this by setting the environment variable WORKING_DIR and reloading the
module (e.g., /dev/shm is a common option) 
Configuring for use of EESSI_USER_INSTALL under /home/bob/eessi
-- To create installations for EESSI, you _must_ have write permissions to
/home/bob/eessi/versions/2023.06/software/linux/x86_64/intel/haswell 
-- You may wish to configure a sources directory for EasyBuild (for example,
via setting the environment variable EASYBUILD_SOURCEPATH) to allow you to
reuse existing sources for packages. 

{EESSI 2023.06} $ echo $EASYBUILD_ALLOW_LOADED_MODULES 
EasyBuild,EESSI-extend
```

**Pull request merged**